app.py

```
@@ -63,34 +65,47 @@ def connect():
+    if serial_connection.target is not None:
+        connected = True
+    
+    # Initialize a new connection if one does not exist
+    if not connected:
+        try:
+            with serial_lock():
+                serial_connection.init_comport(name=name, baudrate=921600, timeout=timeout)
 
-        # send connect opcode
-        serial_connection.connect()
+                if not serial_connection.open_comport(): return "Failed to open comport"
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)